### PR TITLE
#aaujpa

### PR DESCRIPTION
**Pull request by vkirichenko**
[BE] xm-ms-config: get and cache token key implementation

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -176,6 +176,7 @@ repositories {
     mavenLocal()
     mavenCentral()
     jcenter()
+    maven { url "https://oss.sonatype.org/content/repositories/snapshots/" }//TODO remove
```
Comment (vkirichenko) [inline build.gradle:179]: will be removed after commons merging

File: src/main/java/com/icthh/xm/ms/configuration/repository/impl/IdpConfigRepository.java [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+import static com.icthh.xm.commons.domain.idp.IdpConstants.IDP_PUBLIC_SETTINGS_CONFIG_PATH_PATTERN;
+
+/**
+ * This class reads and process IDP clients public configuration for each tenant.
+ * Tenant IDP clients created for each successfully loaded config. If config not loaded or invalid it skipped.
+ */
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class IdpConfigRepository implements RefreshableConfiguration {
```
Comment (sergeysenja1992) [inline src/main/java/com/icthh/xm/ms/configuration/repository/impl/IdpConfigRepository.java:33]: where this repository using?

File: src/main/java/com/icthh/xm/ms/configuration/repository/impl/IdpConfigRepository.java [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    public void onInit(String configKey, String configValue) {
+        updateIdpConfigs(configKey, configValue);
+    }
+
+    @Override
+    public void onRefresh(String updatedKey, String config) {
+        updateIdpConfigs(updatedKey, config);
+    }
+
+    private void updateIdpConfigs(String configKey, String config) {
```
Comment (sergeysenja1992) [inline src/main/java/com/icthh/xm/ms/configuration/repository/impl/IdpConfigRepository.java:70]: pls handle case when config removed, in this case config will be null, and it's case must be handled and all information need to be deleted from memory

File: src/main/java/com/icthh/xm/ms/configuration/repository/impl/IdpConfigRepository.java (hunk 2 above)
Comment (olye0216) [reply src/main/java/com/icthh/xm/ms/configuration/repository/impl/IdpConfigRepository.java:33]: This repository is responsible for processing tenant idp-config-public configuration and is an initiator for retrieving/updating jwks keys from remote IDP and saving them to the file system.

Comment (sergeysenja1992) [reply src/main/java/com/icthh/xm/ms/configuration/repository/impl/IdpConfigRepository.java:33]: it's unused class, i not found place where this class using, pls tell where this class using

File: src/main/java/com/icthh/xm/ms/configuration/repository/impl/IdpConfigRepository.java (hunk 3 above)
Comment (olye0216) [reply src/main/java/com/icthh/xm/ms/configuration/repository/impl/IdpConfigRepository.java:72]: Deletion case handled

**Review by amedvedchuk — APPROVED** [review]

**Review by sergeysenja1992 — APPROVED** [review]

**Pull request merged**